packages/mutation-server-protocol/src/schema.ts

```diff
@@ -0,0 +1,130 @@
+
+export const DiscoverParams = object({
+  /**
+   * The files to run discovery on, or undefined to discover all files in the current project.
+   * A file ending with a `/` indicates a directory. Each path can specify exactly which code blocks to mutate/discover using a mutation range.
+   * This can be done by postfixing your file with `:startLine[:startColumn]-endLine[:endColumn]`.
+   */
+  files: array(string()).optional(),
+});
+
+export type DiscoverParams = z.infer<typeof DiscoverParams>;
+
+const Position = object({
+  line: number(),
+  column: number(),
+});
+
+export const Location = object({
+  start: Position,
+  end: Position,
+});
+
+export type Location = z.infer<typeof Location>;
+
+export const DiscoveredMutant = object({
+  id: string(),
+  location: Location,
+  description: string().optional(),
+  mutatorName: string(),
+  replacement: string().optional(),
+});
+
+export const DiscoverResult = object({
+  mutants: array(DiscoveredMutant),
+});
+
+export type DiscoverResult = z.infer<typeof DiscoverResult>;
+
+export const MutationTestParams = object({
+  /**
+   * The files to run mutation testing on, or undefined to run mutation testing on all files in the current project.
+   * A file ending with a `/` indicates a directory. Each path can specify exactly which code blocks to mutate/discover using a mutation range.
+   * This can be done by postfixing your file with `:startLine[:startColumn]-endLine[:endColumn]`.
+   */
+  files: array(string()).optional(),
+});
+
+export type MutationTestParams = z.infer<typeof MutationTestParams>;
+
+const MutantStatus = enum_([
+  'Killed',
+  'Survived',
+  'NoCoverage',
+  'CompileError',
+  'RuntimeError',
+  'Timeout',
+  'Ignored',
+  'Pending',
+]);
+
+export type MutantStatus = z.infer<typeof MutantStatus>;
+
+export const MutantResult = DiscoveredMutant.extend({
+  /**
+   * The test ids that covered this mutant. If a mutation testing framework doesn't measure this information, it can simply be left out.
+   */
+  coveredBy: array(string()).optional(),
+  /**
+   * The net time it took to test this mutant in milliseconds. This is the time measurement without overhead from the mutation testing framework.
+   */
+  duration: number().optional(),
+  /**
+   * The test ids that killed this mutant. It is a best practice to "bail" on first failing test, in which case you can fill this array with that one test.
+   */
+  killedBy: array(string()).optional(),
+  /**
+   * A static mutant means that it was loaded once at during initialization, this makes it slow or even impossible to test, depending on the mutation testing framework.
+   */
+  static: boolean().optional(),
+  /**
+   * The status of the mutant.
+   */
+  status: MutantStatus,
+  /**
+   * The reason that this mutant has this status. In the case of a killed mutant, this should be filled with the failure message(s) of the failing tests. In case of an error mutant, this should be filled with the error message.
```

Comment:
Are there any conventions / formats / templates for the `statusReason` ? should it be parsable? If so, please add a reference.

Reply:
StatusReason is a free-format text field that it used for example in the popunder in the html report: https://dashboard.stryker-mutator.io/reports/github.com/stryker-mutator/robo-coasters-example/master#mutant/components/robo-coasters.component.js

If you enable runtime errors in the filters and select the first one, it will have a StatusReason. Due to a bug it's not currently very readable in the dashboard, but you get the idea. The StatusReason could be used in a MSP plugin as the 'testrun' output on a mutant.